### PR TITLE
Model required CLI alternative groups

### LINKED ISSUE #4453: Model required alternative groups in generated CLI APIs
PR #4414 makes `Cargo.AddAsync` options optional because the positional dependency can be replaced by `--path` or `--git`. Although no single positional operand is universally required, `cargo add` still requires one member of the dependency-source choice; a parameterless options default allows a guaranteed-invalid bare command.

Improve generator requiredness modeling to represent required alternative groups and keep the options object mandatory when at least one option/operand choice must be supplied.

Acceptance criteria:
- Model CLI usage groups such as `<DEP_ID|--path <PATH>|--git <URI>>`.
- Keep `CargoAddOptions` mandatory while allowing each alternative member to remain individually nullable.
- Add validation or a generated type shape preventing an empty choice group.
- Add scraper and exact argv regression tests for positional, path, git, and empty cases.
- Regenerate Cargo service/interface signatures.

Discovered in #4414.

### PR DESCRIPTION
**Pull request by thomhurst**
Closes #4453.\n\n## Summary\n- model required CLI alternative groups from inline and multi-synopsis usage\n- generate command-level validation while keeping each alternative nullable\n- recover Cargo add's multiline --path and --git options\n- regenerate Cargo add API and exact argv coverage\n\n## Validation\n- OptionsGenerator Release build: passed, 0 warnings\n- OptionsGenerator tests: 1,188 passed\n- Rust Release build: passed, 0 warnings\n- Rust tests: 7 passed\n- Cargo 1.98.0 regeneration: deterministic

## Summary by CodeRabbit

* **New Features**
  * Added support for commands requiring at least one option from a defined set of alternatives.
  * Generated command options now validate that at least one alternative is provided.
  * Improved Cargo command parsing for dependency sources such as paths, Git repositories, registries, and workspace dependencies.
  * Generated service methods now require options when alternative inputs are mandatory.

* **Bug Fixes**
  * Improved handling of command syntax alternatives, aliases, and property name collisions.

### CODE REVIEW

File: tools/ModularPipelines.OptionsGenerator/src/ModularPipelines.OptionsGenerator/Scrapers/Cli/UsageSynopsisParser.cs [hunk 1]
```diff
@@ -180,9 +188,170 @@ private static UsageSynopsisParseResult ParseSynopsis(
...
+            .Select(members => members
+                .Where(member => !commonKeys.Contains(GetAlternativeMemberKey(member)))
+                .ToArray())
+            .ToArray();
+        if (branchMembers.Any(static members => members.Length == 0))
+        {
+            return [];
+        }
+
+        var alternatives = DistinctAlternativeMembers(branchMembers.SelectMany(static members => members));
```
[the remark this answers is not in the document]
Comment (thomhurst) [reply tools/ModularPipelines.OptionsGenerator/src/ModularPipelines.OptionsGenerator/Scrapers/Cli/UsageSynopsisParser.cs:315]: Fixed in `3c4fa22ed4`. Required-alternative inference now declines any branch containing anything other than one required member, so conjunctions cannot be flattened into an `||` group. Added `Does_Not_Flatten_Conjunctive_Alternative_Branches` regression coverage for the reported two-member forms.

**Comment by thomhurst** [comment]
Addressed both review findings in `3c4fa22ed4`:

- Cargo option parsing now uses one strict declaration scan across the full help document, so `Source options:`, `Package Selection:`, and other named sections share the same path instead of required-alternative-only recovery.
- Inline descriptions seed the shared continuation parser; regressions assert `--path`, `--git`, `--registry`, and a non-`Options:` `--workspace` entry plus their descriptions.
- Multi-member alternative branches are conservatively excluded from OR-group inference.

Validation: OptionsGenerator and Rust Release builds; `UsageSynopsisParserTests` 75/75; `CliScraperTraversalTests` 43/43; `CargoOptionsTests` 6/6. Full coverage run hit the repository guard's fixed 2 GB limit (2329 MB), so CI owns that full check.

@claude review

File: tools/ModularPipelines.OptionsGenerator/src/ModularPipelines.OptionsGenerator/Scrapers/Cli/UsageSynopsisParser.cs [hunk 2]
```diff
@@ -180,9 +188,170 @@ private static UsageSynopsisParseResult ParseSynopsis(
...
+
+            var alternatives = SplitTopLevelAlternatives(TrimWrapper(normalizedToken));
+            if (alternatives.Count <= 1)
+            {
+                continue;
+            }
+
+            var alternativeMembers = alternatives
+                .Select(alternative => GetRequiredAlternativeMembers(
+                    ParseOperandTokens(Tokenize(alternative), phase)))
```
[the remark this answers is not in the document]
Comment (thomhurst) [reply tools/ModularPipelines.OptionsGenerator/src/ModularPipelines.OptionsGenerator/Scrapers/Cli/UsageSynopsisParser.cs]: Fixed in 597d8ec01d. Synopsis parsing now retains presence-only option switches as required-alternative members. Added the brew services stop (<formula>|--all) regression.

File: tools/ModularPipelines.OptionsGenerator/src/ModularPipelines.OptionsGenerator/Models/CliRequiredAlternativeGroup.cs [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+
+/// <summary>
+/// A command-level constraint requiring at least one generated option or operand.
+/// </summary>
+public sealed record CliRequiredAlternativeGroup
+{
+    /// <summary>
+    /// Generated property names participating in the choice.
+    /// </summary>
+    public required IReadOnlyList<string> PropertyNames { get; init; }
```
[the remark this answers is not in the document]
Comment (thomhurst) [reply tools/ModularPipelines.OptionsGenerator/src/ModularPipelines.OptionsGenerator/Models/CliRequiredAlternativeGroup.cs]: Fixed in 597d8ec01d. Required alternatives now retain stable option-switch or positional phase/position identity; collision resolution maps each identity to its final property name. Added scraper and generated-validation regressions for colliding Filename option/operand members.

**Comment by thomhurst** [comment]
Addressed the remaining required-alternative review findings in 597d8ec01d:

- presence-only switches such as brew services stop --all remain alternative members
- option/operand alternatives retain stable identity through property collision resolution
- colliding Filename members now validate both Filename and FilenameArgument

Validation: OptionsGenerator tests 1192/1192 passed; scoped whitespace verification passed; OptionsGenerator Release build passed with 0 warnings/errors. The prior Ubuntu failure was the unrelated ExternalMetadata_DoesNotRootCollectibleConsumerAssembly GC assertion.

File: tools/ModularPipelines.OptionsGenerator/src/ModularPipelines.OptionsGenerator/Scrapers/Cli/UsageSynopsisParser.cs [hunk 4]
```diff
@@ -180,15 +188,182 @@ private static UsageSynopsisParseResult ParseSynopsis(
...
+    private static IReadOnlyList<UsageRequiredAlternativeMember> GetRequiredAlternativeMembers(
+        ParsedOperands operands) =>
+        DistinctAlternativeMembers(
+            operands.Arguments
+                .Where(static argument => argument.IsRequired)
+                .Select(static argument => argument.AssociatedOptionSwitch is { } optionSwitch
+                    ? new UsageRequiredAlternativeMember { OptionSwitch = optionSwitch }
+                    : new UsageRequiredAlternativeMember { PositionalPropertyName = argument.PropertyName })
+                .Concat(operands.OptionSwitches.Select(static optionSwitch =>
+                    new UsageRequiredAlternativeMember { OptionSwitch = optionSwitch })));
```
[the remark this answers is not in the document]
Comment (thomhurst) [reply tools/ModularPipelines.OptionsGenerator/src/ModularPipelines.OptionsGenerator/Scrapers/Cli/UsageSynopsisParser.cs]: Fixed in ac6e1aa1ef. Synopsis parsing now tracks required switches separately, so bracketed switches remain available for collision association but cannot become required-alternative members. Added the reported two-synopsis regression; parser suite 77/77 and full OptionsGenerator suite 1193/1193 pass.

File: tools/ModularPipelines.OptionsGenerator/src/ModularPipelines.OptionsGenerator/Scrapers/Cli/UsageSynopsisParser.cs [hunk 5]
```diff
@@ -180,15 +188,185 @@ private static UsageSynopsisParseResult ParseSynopsis(
...
+            : [];
+    }
+
+    private static bool SupportsRequiredAlternativeInference(IEnumerable<string> operandTokens)
+    {
+        foreach (var token in CollapseAlternatives(operandTokens))
+        {
+            if (IsNonOperandSyntax(token))
+            {
+                continue;
```
[the remark this answers is not in the document]
Comment (thomhurst) [reply tools/ModularPipelines.OptionsGenerator/src/ModularPipelines.OptionsGenerator/Scrapers/Cli/UsageSynopsisParser.cs:341]: Implemented in `d16b1afe6d`. `SupportsRequiredAlternativeInference` now recognizes parsed option switches before the non-operand filter, so a standalone required switch remains an eligible cross-synopsis branch. Added `tool clean <TARGET>` / `tool clean --all` regression coverage.

File: tools/ModularPipelines.OptionsGenerator/src/ModularPipelines.OptionsGenerator/Scrapers/Cli/UsageSynopsisParser.cs [hunk 6]
```diff
@@ -180,15 +188,185 @@ private static UsageSynopsisParseResult ParseSynopsis(
...
+            var alternatives = SplitTopLevelAlternatives(TrimWrapper(normalizedToken));
+            if (alternatives.Count <= 1)
+            {
+                continue;
+            }
+
+            var alternativeMembers = alternatives
+                .Select(alternative => GetRequiredAlternativeMembers(
+                    ParseOperandTokens(Tokenize(alternative), phase)))
+                .ToArray();
```
[the remark this answers is not in the document]
Comment (thomhurst) [reply tools/ModularPipelines.OptionsGenerator/src/ModularPipelines.OptionsGenerator/Scrapers/Cli/UsageSynopsisParser.cs:246]: Implemented in `d16b1afe6d`. Option extraction now retains the switch prefix before `=` while marking its value inline, so `(<FILE>|--data=<TEXT>)` produces required members `File` and `--data` without attaching the following operand. Added direct regression coverage.

**Comment by thomhurst** [comment]
Addressed the full latest blocking review in `d16b1afe6d`:

- standalone switches now participate in cross-synopsis required alternatives
- attached-value switches retain canonical option identity
- parenthesized option aliases split into real switches instead of a pipe-joined literal
- Cargo option descriptions stop at the first blank-line paragraph boundary
- required-alternative validation uses the shared collection-shape detector, including `IReadOnlyList<T>`
- required-alternative option lookup is shared by scraper resolution and collision resolution, including short forms and case-insensitive matching

Validation: OptionsGenerator tests 1207/1207; Release build 0 warnings/errors; scoped format verification passed; `git diff --check` passed.

**Comment by thomhurst** [comment]
Addressed both CodeFactor findings in 4aa269482d: ParseOperandTokens now delegates option/final-operand handling to focused helpers, and the indexable fallback switch uses [0] instead of LINQ FirstOrDefault. Validation: UsageSynopsisParserTests 83/83; full OptionsGenerator suite 1207/1207; Release generator solution build 0 warnings/errors; scoped whitespace verification passed.

File: src/ModularPipelines.Rust/Options/CargoAddOptions.Generated.cs [hunk 7]
```diff
@@ -19,12 +20,33 @@ namespace ModularPipelines.Rust.Options;
...
+    /// Filesystem path to local crate to add
+    /// </summary>
+    [CliOption("--path")]
+    public string? Path { get; set; }
+
+    /// <summary>
+    /// Git repository location Without any other information, cargo will use latest commit on the main branch.
+    /// </summary>
+    [CliOption("--git")]
+    public string? Git { get; set; }
```
[the remark this answers is not in the document]
Comment (thomhurst) [reply src/ModularPipelines.Rust/Options/CargoAddOptions.Generated.cs:149]: Fixed in cac5d13fdf. Regenerated Cargo authoritatively from current Cargo 1.98.0 help; all 16 option classes now contain the complete current option sets, including Cargo add Registry and Rename. The installed current CLI does not expose --workspace, so that fixture-only switch is intentionally absent under the repository source-of-truth rule. Generation was deterministic across two runs.

File: tools/ModularPipelines.OptionsGenerator/src/ModularPipelines.OptionsGenerator/Scrapers/Cli/UsageSynopsisParser.cs [hunk 8]
```diff
@@ -110,6 +110,13 @@ public static UsageSynopsisParseResult Parse(
             PositionalArguments = RelaxArgumentsMissingFromAlternatives(
                 selected.PositionalArguments,
                 requirednessCandidates),
+            RequiredAlternativeGroups =
+            [
+                .. selected.RequiredAlternativeGroups,
+                .. GetCrossSynopsisRequiredAlternativeGroups(
+                    requirednessCandidates,
+                    selected.PositionalArguments),
+            ],
```
[the remark this answers is not in the document]
Comment (thomhurst) [reply tools/ModularPipelines.OptionsGenerator/src/ModularPipelines.OptionsGenerator/Scrapers/Cli/UsageSynopsisParser.cs:125]: Fixed in cac5d13fdf. Removing command-group placeholders now also discards any required-alternative group containing the removed positional member. Added an end-to-end traversal regression for `fake parent <command>` / `fake parent --all`, proving both parent and child remain generated.

**Comment by thomhurst** [comment]
Addressed both fresh review findings in `cac5d13fdf`: command-placeholder alternative groups no longer skip executable parents, and Cargo was authoritatively regenerated from current 1.98.0 help (16 complete option classes; deterministic second run). Validation: OptionsGenerator 1208/1208; Rust 7/7; both Release solutions build with 0 warnings/errors; changed-file whitespace checks pass. Full Rust formatting only reports unrelated pre-existing `src/ModularPipelines/Caching/ZipCentralDirectory.cs` whitespace.

File: tools/ModularPipelines.OptionsGenerator/src/ModularPipelines.OptionsGenerator/Scrapers/Cli/UsageSynopsisParser.cs [hunk 9]
```diff
@@ -180,15 +188,190 @@ private static UsageSynopsisParseResult ParseSynopsis(
...
+        {
+            commonKeys.IntersectWith(members.Select(GetAlternativeMemberKey));
+        }
+
+        var branchMembers = candidateMembers
+            .Select(members => members
+                .Where(member => !commonKeys.Contains(GetAlternativeMemberKey(member)))
+                .ToArray())
+            .ToArray();
+        if (branchMembers.Any(static members => members.Length != 1))
```
[the remark this answers is not in the document]
Comment (thomhurst) [reply tools/ModularPipelines.OptionsGenerator/src/ModularPipelines.OptionsGenerator/Scrapers/Cli/UsageSynopsisParser.cs:310]: Fixed in ffab98fa9e. Cross-synopsis inference now canonicalizes syntactic two-switch short/long alias groups (for example `-a|--all`) before counting branch members. Public parsed spellings remain intact, and true multi-switch conjunctions remain rejected. Added the exact regression; full generator suite passes 1209/1209.

**Comment by thomhurst** [comment]
Follow-up `ffab98fa9e` fixes short/long alias counting in cross-synopsis alternatives. Validation: OptionsGenerator 1209/1209; Release build 0 warnings/errors; scoped whitespace verification passed.

File: tools/ModularPipelines.OptionsGenerator/src/ModularPipelines.OptionsGenerator/Scrapers/Cli/UsageSynopsisParser.cs [hunk 10]
```diff
@@ -961,27 +1238,54 @@ private static bool IsPhaseControlToken(string token)
...
+            .ToArray();
+    }
+
+    private static string? GetOptionSwitch(string alternative)
+    {
+        var content = TrimControlWrappers(alternative);
         var endIndex = content.IndexOfAny([' ', '\t', '=']);
-        optionSwitch = content.TrimEnd(',', ':');
+        var optionSwitch = (endIndex < 0 ? content : content[..endIndex])
+            .TrimEnd('[', ',', ':');
```
[the remark this answers is not in the document]
Comment (thomhurst) [reply tools/ModularPipelines.OptionsGenerator/src/ModularPipelines.OptionsGenerator/Scrapers/Cli/UsageSynopsisParser.cs:1384]: Fixed at rebased head fd94c8c775. Grouped branches now expand multiple switches only when every nested token is an option. Conjunctive branches therefore contain multiple members and remain unsupported instead of truncating to the first flag. Added the exact regression.

File: tools/ModularPipelines.OptionsGenerator/src/ModularPipelines.OptionsGenerator/Scrapers/Cli/UsageSynopsisParser.cs [hunk 11]
```diff
@@ -110,6 +110,13 @@ public static UsageSynopsisParseResult Parse(
             PositionalArguments = RelaxArgumentsMissingFromAlternatives(
                 selected.PositionalArguments,
                 requirednessCandidates),
+            RequiredAlternativeGroups =
+            [
+                .. selected.RequiredAlternativeGroups,
+                .. GetCrossSynopsisRequiredAlternativeGroups(
+                    requirednessCandidates,
+                    selected.PositionalArguments),
```
[the remark this answers is not in the document]
Comment (thomhurst) [reply tools/ModularPipelines.OptionsGenerator/src/ModularPipelines.OptionsGenerator/Scrapers/Cli/UsageSynopsisParser.cs:124]: Fixed at rebased head fd94c8c775. A selected inline group is retained only when every same-command requiredness candidate contains one of its members. A separate stdin form now removes the narrower file-or-URL constraint. Added regression coverage.

**Comment by thomhurst** [comment]
@claude review

**Comment by thomhurst** [comment]
@codex review

**Comment by thomhurst** [comment]
Completed in da695fafdb. Inline alternative branches now collapse short/long option aliases through the same canonicalization used by cross-synopsis inference before cardinality validation. `(<TARGET>|(-a|--all))` therefore generates the required `Target | --all` group while keeping storage optional. Parser suite passes 91/91; generator Release build has 0 warnings/errors.

File: tools/ModularPipelines.OptionsGenerator/src/ModularPipelines.OptionsGenerator/Scrapers/Cli/CliScraperBase.cs [hunk 12]
```diff
@@ -523,6 +523,7 @@ private bool ShouldSkipCommand(
             {
                 HasOperandTakingUsage = usage.HasOperandTokens,
                 UsagePositionalArguments = usage.PositionalArguments,
+                RequiredAlternativeGroups = ResolveRequiredAlternativeGroups(command, usage),
```
[the remark this answers is not in the document]
Comment (thomhurst) [reply tools/ModularPipelines.OptionsGenerator/src/ModularPipelines.OptionsGenerator/Scrapers/Cli/CliScraperBase.cs:526]: Fixed in `cf1ca70668`. Inferred groups now resolve atomically: if any synopsis member is inherited, global, filtered, or otherwise unavailable, that inferred constraint is discarded while the command remains generated. Added an end-to-end regression.

File: tools/ModularPipelines.OptionsGenerator/src/ModularPipelines.OptionsGenerator/Generators/OptionsClassGenerator.cs [hunk 13]
```diff
@@ -56,6 +56,7 @@ private static string GenerateOptionsClass(CliCommandDefinition command, CliTool
 
         sb.AppendLine("{");
         GenerateProperties(sb, command, positionalArguments, existingPropertyNames);
+        GenerateRequiredAlternativeValidation(sb, command, positionalArguments);
```
[the remark this answers is not in the document]
Comment (thomhurst) [reply tools/ModularPipelines.OptionsGenerator/src/ModularPipelines.OptionsGenerator/Generators/OptionsClassGenerator.cs:59]: No change needed. Generator-wide failure is intentional: `GenerateForToolAsync` creates and validates the complete tool output in memory before writes or stale cleanup. Per-command isolation would permit a malformed required group to silently produce a partial API; fail-fast preserves atomic generation.

File: tools/ModularPipelines.OptionsGenerator/src/ModularPipelines.OptionsGenerator/Models/CliOptionDefinition.cs [hunk 14]
```diff
@@ -79,6 +79,14 @@ internal static bool TryGetCollectionShape(string cSharpType, out bool isCollect
         return resolution.IsResolved;
     }
 
+    internal static int FindIndexBySwitch(
```
[the remark this answers is not in the document]
Comment (thomhurst) [reply tools/ModularPipelines.OptionsGenerator/src/ModularPipelines.OptionsGenerator/Models/CliOptionDefinition.cs:82]: Fixed in `cf1ca70668`. Switch identity and lookup now use ordinal case-sensitive comparison, so `-o` and `-O` remain distinct through parsing, deduplication, and collision resolution. Added parser and direct lookup regressions.

File: tools/ModularPipelines.OptionsGenerator/src/ModularPipelines.OptionsGenerator/Scrapers/Cli/CliScraperBase.cs [hunk 15]
```diff
@@ -786,6 +787,80 @@ protected virtual UsageSynopsisParseResult NormalizeUsageSynopsis(
...
+            {
+                Members = group.Members
+                    .Select(member => ResolveRequiredAlternativeMember(command, member))
+                    .DistinctBy(GetRequiredAlternativeIdentity, StringComparer.OrdinalIgnoreCase)
+                    .ToArray(),
+            }),
+        ];
+    }
+
+    private static CliRequiredAlternativeMember ResolveRequiredAlternativeMember(
```
[the remark this answers is not in the document]
Comment (thomhurst) [reply tools/ModularPipelines.OptionsGenerator/src/ModularPipelines.OptionsGenerator/Scrapers/Cli/CliScraperBase.cs]: Fixed in `cf1ca70668`. Unresolvable synopsis-inferred groups are now omitted atomically instead of throwing through `TryParseCommandAsync`, so the command is retained. End-to-end regression covers an unavailable global switch.

File: tools/ModularPipelines.OptionsGenerator/src/ModularPipelines.OptionsGenerator/Generators/OptionsClassGenerator.cs [hunk 16]
```diff
@@ -183,16 +185,97 @@ private static HashSet<string> GenerateClassDeclaration(
...
+            sb.AppendLine($"        if (!({presenceExpression}))");
+            sb.AppendLine("        {");
+            sb.AppendLine($"            yield return new ValidationResult({GeneratorUtils.FormatStringLiteral(message)}, [{memberNames}]);");
+            sb.AppendLine("        }");
+        }
+        sb.AppendLine("    }");
+        sb.AppendLine();
+    }
+
+    private static string GetPresenceExpression(
```
[the remark this answers is not in the document]
Comment (thomhurst) [reply tools/ModularPipelines.OptionsGenerator/src/ModularPipelines.OptionsGenerator/Generators/OptionsClassGenerator.cs:243]: No change needed. Tool generation is intentionally atomic, not per-command best-effort: all files are generated and validated before disk mutation. An invalid hand-authored/cached group must abort the tool instead of silently emitting incomplete generated output.

File: tools/ModularPipelines.OptionsGenerator/src/ModularPipelines.OptionsGenerator/Scrapers/Cli/UsageSynopsisParser.cs [hunk 17]
```diff
@@ -1006,27 +1304,70 @@ private static bool IsPhaseControlToken(string token)
...
+        {
+            return nestedSwitches;
         }
 
+        return GetOptionSwitch(alternative) is { } optionSwitch
+            ? [optionSwitch]
+            : [];
+    }
+
+    private static string? GetOptionSwitch(string alternative)
```
[the remark this answers is not in the document]
Comment (thomhurst) [reply tools/ModularPipelines.OptionsGenerator/src/ModularPipelines.OptionsGenerator/Scrapers/Cli/UsageSynopsisParser.cs:1379]: Fixed in `cf1ca70668`. Nested-group `TryGetOptionSwitch` again accepts only standalone switches; attached/trailing-value extraction remains scoped to required-alternative parsing. Regression proves `<[--format=VALUE] file>` does not associate `file` with `--format`.

File: tools/ModularPipelines.OptionsGenerator/src/ModularPipelines.OptionsGenerator/Scrapers/Cli/CargoCliScraper.cs [hunk 18]
```diff
@@ -193,33 +193,15 @@ private static bool IsValidCommand(string name)
     /// Format: -V, --version    Print version info
     ///         --list           List installed commands
     /// </summary>
-    private List<CliOptionDefinition> ParseOptions(string helpText)
+    private static List<CliOptionDefinition> ParseOptions(string helpText)
```
[the remark this answers is not in the document]
Comment (thomhurst) [reply tools/ModularPipelines.OptionsGenerator/src/ModularPipelines.OptionsGenerator/Scrapers/Cli/CargoCliScraper.cs:196]: Fixed in `cf1ca70668`. Cargo option parsing now runs only inside top-level option/selection sections, retaining `Source options` and `Package Selection` while ignoring option-shaped lines under `Examples`. Regression added to the Cargo traversal fixture.

File: tools/ModularPipelines.OptionsGenerator/src/ModularPipelines.OptionsGenerator/Scrapers/Cli/UsageSynopsisParser.cs [hunk 19]
```diff
@@ -180,15 +200,243 @@ private static UsageSynopsisParseResult ParseSynopsis(
...
+                continue;
+            }
+
+            var alternativeMembers = alternatives
+                .Select(alternative => CollapseOptionAliases(
+                    GetRequiredAlternativeMembers(
+                        ParseOperandTokens(Tokenize(alternative), phase)),
+                    alternative))
+                .ToArray();
+            if (alternativeMembers.Any(static members => members.Count != 1))
```
[the remark this answers is not in the document]
Comment (thomhurst) [reply tools/ModularPipelines.OptionsGenerator/src/ModularPipelines.OptionsGenerator/Scrapers/Cli/UsageSynopsisParser.cs:247]: Fixed in `cf1ca70668`. Comma-separated short/long aliases are normalized before branch parsing and collapsed to one generated option identity. Regression covers `(<TARGET>|-a, --all)` and guards whitespace conjunctions separately.

File: tools/ModularPipelines.OptionsGenerator/src/ModularPipelines.OptionsGenerator/Scrapers/Cli/UsageSynopsisParser.cs [hunk 20]
```diff
@@ -1006,27 +1304,70 @@ private static bool IsPhaseControlToken(string token)
...
         }
 
-        optionSwitch = "";
-        return false;
+        return null;
+    }
+
+    private static string? SelectPreferredOptionSwitch(IReadOnlyList<string> optionSwitches) =>
+        optionSwitches.FirstOrDefault(static optionSwitch => optionSwitch.StartsWith("--", StringComparison.Ordinal))
+        ?? (optionSwitches.Count > 0 ? optionSwitches[0] : null);
```
[the remark this answers is not in the document]
Comment (thomhurst) [reply tools/ModularPipelines.OptionsGenerator/src/ModularPipelines.OptionsGenerator/Scrapers/Cli/UsageSynopsisParser.cs:1397]: Fixed in `cf1ca70668`. The broader attached-value extraction remains in required-alternative parsing, while nested-group association now uses strict standalone-switch recognition. The exact attached-value nested-group regression is included.

File: tools/ModularPipelines.OptionsGenerator/src/ModularPipelines.OptionsGenerator/Generators/InheritedPropertyCollisionResolver.cs [hunk 21]
```diff
@@ -113,6 +127,51 @@ private static CliCommandDefinition ResolveCommand(
...
+            }
+
+            return options[optionIndex].PropertyName;
+        }
+
+        if (member.PositionalArgumentPhase is { } phase
+            && member.PositionalArgumentPositionIndex is { } positionIndex
+            && member.OptionSwitch is null)
+        {
+            var argumentIndex = Enumerable.Range(0, command.PositionalArguments.Count).FirstOrDefault(index =>
```
[the remark this answers is not in the document]
Comment (thomhurst) [reply tools/ModularPipelines.OptionsGenerator/src/ModularPipelines.OptionsGenerator/Generators/InheritedPropertyCollisionResolver.cs:153]: No change needed. The two lookups intentionally identify different stages: scraper resolution matches the inferred pre-collision positional property name; collision resolution matches the stable phase/position identity after renaming. A shared phase/position helper would not replace the pre-collision lookup.

File: tools/ModularPipelines.OptionsGenerator/src/ModularPipelines.OptionsGenerator/Scrapers/Cli/UsageSynopsisParser.cs [hunk 22]
```diff
@@ -180,15 +200,243 @@ private static UsageSynopsisParseResult ParseSynopsis(
...
+                candidate.UnparsedOperandTokens,
+                candidate.RequiredOptionSwitches))
+            .Concat(candidate.RequiredAlternativeGroups.SelectMany(static candidateGroup =>
+                candidateGroup.Members))
+            .Select(GetAlternativeMemberKey)
+            .ToHashSet(StringComparer.OrdinalIgnoreCase);
+
+        return group.Members.Any(member =>
+            candidateMemberKeys.Contains(GetAlternativeMemberKey(member)));
+    }
```
[the remark this answers is not in the document]
Comment (thomhurst) [reply tools/ModularPipelines.OptionsGenerator/src/ModularPipelines.OptionsGenerator/Scrapers/Cli/UsageSynopsisParser.cs]: Fixed in `cf1ca70668`. Required member keys are computed once per candidate synopsis before filtering inline groups, removing repeated member derivation from the groups-by-candidates loop.
